### PR TITLE
HTTP Server: Add support for Gzip Compression + Reuse ByteArray

### PR DESCRIPTION
**Pull request by erebe**
Compatible with java 1.6
  + Add support for gzip compression
  + Reuse ByteArray instead of requesting a new one for each request

### CODE REVIEW

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java [hunk 1]
```diff
@@ -51,13 +63,33 @@ public void handle(HttpExchange t) throws IOException {
...
         }
 
     }
 
+    protected static boolean shouldUseCompression(HttpExchange exchange) {
+        List<String> encodings = exchange.getRequestHeaders().get("Accept-Encoding");
+        if (encodings == null) return false;
+
+        for (String encoding: encodings) {
+            if(encoding.contains("gzip")) {
```
Comment (brian-brazil) [inline simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java]: space after if

**Comment by erebe** [comment]
Updated for the space, and understood for until next week

**Comment by erebe** [comment]
gentle bump

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java [hunk 2]
```diff
@@ -27,8 +30,16 @@
  * </pre>
  * */
 public class HTTPServer {
+    static class LocalByteArray extends ThreadLocal<ByteArrayOutputStream> {
```
Comment (brian-brazil) [inline simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java]: I'd mark this explicitly private

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java [hunk 3]
```diff
@@ -51,13 +63,33 @@ public void handle(HttpExchange t) throws IOException {
...
             t.close();
         }
 
     }
 
+    protected static boolean shouldUseCompression(HttpExchange exchange) {
+        List<String> encodings = exchange.getRequestHeaders().get("Accept-Encoding");
+        if (encodings == null) return false;
+
+        for (String encoding: encodings) {
```
Comment (brian-brazil) [inline simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java]: space before :

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java [hunk 4]
```diff
@@ -51,13 +63,33 @@ public void handle(HttpExchange t) throws IOException {
...
+            if (encoding.contains("gzip")) {
```
Comment (brian-brazil) [inline simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java]: To be safe you should probably split this and then do an equality check

**Comment by erebe** [comment]
Done

**Comment by erebe** [comment]
Arf not yet `com.sun.deploy.util does not exist`
Fixing

**Comment by erebe** [comment]
I witched to user String.split aven if it used a regex and it is overkill as use it also below in parseQuery

**Comment by erebe** [comment]
Waving

**Comment by brian-brazil** [comment]
Thanks!

**Comment by erebe** [comment]
Thank you for your time for the review :)

**Pull request merged**